### PR TITLE
Animations

### PR DESCRIPTION
**Pull request by matejsemancik**
### library module
Allow to configure:
- enable / disable animation flag
- animation duration
- animation interpolator

Moved defaults to companion object.

### sample module
Added animation controls:
- enable switch
- duration seeker
- interpolator chooser

### CODE REVIEW

File: library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt [hunk 1]
```diff
@@ -48,7 +58,7 @@ class DonutProgressView @JvmOverloads constructor(
      * Eg. when one line has 50% of total graph length,
      * setting this to 0.5f will result in that line being animated to 25% of total graph length.
      */
-    var masterProgress: Float = 1f
+    var masterProgress: Float = DEFAULT_MASTER_PROGRESS
         set(value) {
             if (value !in (0f..1f)) {
```
Comment (thefuntasty-ops) [inline library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt#L63"></span>Parentheses in (0f..1f) are unnecessary and can be replaced with: 0f..1f</td>
    </tr>
  </tbody>
</table>

File: library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt [hunk 2]
```diff
@@ -296,8 +345,8 @@ class DonutProgressView @JvmOverloads constructor(
         animationEnded: (() -> Unit)? = null
     ): ValueAnimator {
         return ValueAnimator.ofFloat(line.length, to).apply {
-            duration = ANIM_DURATION_MS
-            interpolator = DecelerateInterpolator(INTERPOLATOR_FACTOR)
+            duration = if (animationEnabled) animationDurationMs else 0L
+            interpolator = animationInterpolator
             addUpdateListener {
                 line.length = it.animatedValue as Float
```
Comment (thefuntasty-ops) [inline library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt#L351"></span>it.animatedValue cannot be safely cast to Float.</td>
    </tr>
  </tbody>
</table>

File: sample/src/main/kotlin/com/thefuntasty/donutsample/ui/playground/PlaygroundActivity.kt [hunk 3]
```diff
@@ -2,6 +2,7 @@ package com.thefuntasty.donutsample.ui.playground
 
 import android.os.Bundle
```
Comment (thefuntasty-ops) [inline sample/src/main/kotlin/com/thefuntasty/donutsample/ui/playground/PlaygroundActivity.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/sample/src/main/kotlin/com/thefuntasty/donutsample/ui/playground/PlaygroundActivity.kt#L3"></span>Imports must be ordered in lexicographic order without any empty lines in-between</td>
    </tr>
  </tbody>
</table>

**Comment by thefuntasty-ops** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          12 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L13">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L13</a> - Constructor parameter names should match the pattern: [a-z][A-Za-z\d]*</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L14">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L14</a> - Constructor parameter names should match the pattern: [a-z][A-Za-z\d]*</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L15">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L15</a> - Constructor parameter names should match the pattern: [a-z][A-Za-z\d]*</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L16">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L16</a> - Constructor parameter names should match the pattern: [a-z][A-Za-z\d]*</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L17">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L17</a> - Constructor parameter names should match the pattern: [a-z][A-Za-z\d]*</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L18">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L18</a> - Constructor parameter names should match the pattern: [a-z][A-Za-z\d]*</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L19">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L19</a> - Constructor parameter names should match the pattern: [a-z][A-Za-z\d]*</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L120">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressLine.kt#L120</a> - Parentheses in (pathLen.toDouble() * length) are unnecessary and can be replaced with: pathLen.toDouble() * length</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt#L21">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt#L21</a> - Class DonutProgressView is too large. Consider splitting it into smaller pieces.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt#L152">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt#L152</a> - The caught exception is swallowed. The original exception could be lost.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt#L324">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt#L324</a> - Functions with exact one statement, the return statement, can be rewritten with ExpressionBodySyntax.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/thefuntasty/donut/blob/828b548c8186e32727ef08c33dcf88e758e87083/library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt#L407">library/src/main/kotlin/com/thefuntasty/donut/DonutProgressView.kt#L407</a> - Private function dpToPx is unused.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by skywall** [comment]
Stav?

**Comment by matejsemancik** [comment]
@skywall need to resolve #11 first

**Pull request closed without merging**